### PR TITLE
Adding timeout to Downloader.getResponse

### PR DESCRIPTION
**Pull request by anshul0708**

### CODE REVIEW

File: src/downloader.ts [hunk 1]
```diff
@@ -82,14 +85,21 @@ export class Downloader {
...
+            request.setTimeout(Downloader.timeout, response => {
+                let err : Error = new RequestTimeoutError('Request for file:'+parsedURL.pathname+' was timed out after '+Downloader.timeout+' ms');
+                callback(err,undefined);
+            });
         } else if (parsedURL.protocol === 'http:') {
-            Downloader.http.get(fileURL, response => {
+            const request = Downloader.http.get(fileURL, response => {
                 callback(undefined, response);
             });
+            request.setTimeout(Downloader.timeout, response => {
```
Comment (nyxathid) [inline src/downloader.ts]: Can we have an optional timeout parameter, please? Just add if after callback. If it is not provided, we use the default 1000 value. The CC index is very slow in general and sometimes takes more than 10 sec to respond.

Comment (lgfrbcsgo) [inline src/downloader.ts]: Hm, true. I didn't think about that. The signature should look like this then to ensure backwards compatibly
```
getResponse(fileURL : string, callback? : (err? : Error, resp? : IncomingMessage, timeout? : number) => void)
```

Comment (nyxathid) [inline src/downloader.ts]: Yes, exactly!

**Comment by nyxathid** [comment]
added timeout as parameter, please review

**Pull request merged**